text/0018-bee-bundle/0018-bee-bundle.md

```diff
@@ -0,0 +1,308 @@
++ Feature name: `bee-bundle`
++ Start date: 2019-10-02
++ RFC PR: [iotaledger/bee-rfcs#18](https://github.com/iotaledger/bee-rfcs/pull/18)
++ Bee issue: [iotaledger/bee#63](https://github.com/iotaledger/bee/issues/63)
+
+# Summary
+
+This RFC is based on [`transaction-module`]().
+<!-- TODO FIx link to RFC#3 -->
+
+The smallest communication unit in the IOTA protocol is the transaction. Everything, including payment settlements
+and/or plain data, is propagated through the IOTA network in transactions.
+
+A transaction is `2673` trytes and the part available to the user is `2187` trytes. This part holds a signature in case
+of a payment settlement and plain data otherwise. Since it has a limited size, a user often needs more than one
+transaction to fulfil their operation, for example signatures with security level `2` or `3` don't fit in a single
+transaction and user-provided data may exceed the allowance so they need to be fragmented across multiple transactions.
```

Comment:
This is gonna be a bit controversial, but I feel this is the only point in time where we can change a fundamental term:

Isn't essentially the `bundle` what should be called a `transaction` according to the database definition of a transaction where a transaction is an *atomic* operation. But in IOTA's case it is the bundle that is atomic, not the transaction, right?
Atomicity: A transaction must be fully complete, saved (committed) or completely undone (rolled back). A sale in a retail store database illustrates a scenario which explains atomicity, e.g., the sale consists of an inventory reduction and a record of incoming cash. Both either happen together or do not happen - it's all or nothing.

Likewise either all the transactions in a bundle get confirmed, or none of them.

The reasoning for bringing this up is again looking on the proposal from the outside as someone who might not be familiar with IOTA terms, and making its terminology most accessible to newcomers. 

It doesn't sound as sexy but I would propose the terms `Transaction` (former bundle) and `Transaction Fragment` (former transaction), while the `Signature Or Message Fragment` is renamed to `Payload`.

If I have made an essential mistake here, then please correct me. Wouldn't be the first time :trollface:

Reply:
You are 100% right and I've had this thought too. We could have `Transaction` and `Statement` to match the database metaphor. Unfortunately you're  also right saying it is controversial. Not entirely sure we can change terms at this point ?